### PR TITLE
Added router tests for pools

### LINKED ISSUE #912: Write tests for pool routers and process_pools

### PR DESCRIPTION
**Pull request by TheBits**
Fixes #912

### CODE REVIEW

File: src/dstack/_internal/server/background/tasks/process_pools.py [hunk 1]
```diff
@@ -110,24 +111,29 @@ async def check_shim(instance_id: UUID) -> None:
         else:
             logger.debug("check instance %s status: shim health: %s", instance.name, health)
 
-            instance.fail_count += 1
-            instance.fail_reason = health.reason
+            if instance.termination_deadline is None:
+                instance.termination_deadline = get_current_datetime() + timedelta(minutes=20)
```
Comment (Egor-S) [inline src/dstack/_internal/server/background/tasks/process_pools.py]: Let's make a constant from timedelta 20 min

File: src/dstack/_internal/server/background/tasks/process_pools.py [hunk 2]
```diff
@@ -110,24 +111,29 @@ async def check_shim(instance_id: UUID) -> None:
...
+                deadline = instance.termination_deadline.replace(tzinfo=datetime.timezone.utc)
+                if get_current_datetime() > deadline:
                     instance.status = InstanceStatus.TERMINATING
+                    instance.termination_reason = "Termination deadline"
                     logger.warning("mark instance %s as TERMINATED", instance.name)
 
             if instance.status == InstanceStatus.STARTING and instance.started_at is not None:
-                STARTING_TIMEOUT = 10 * 60  # 10 minutes
-                starting_time_threshold = instance.started_at + timedelta(seconds=STARTING_TIMEOUT)
+                starting_timeout = 10 * 60  # 10 minutes
```
Comment (Egor-S) [inline src/dstack/_internal/server/background/tasks/process_pools.py]: Why can't we declare it at the top of the file?

File: src/dstack/_internal/server/background/tasks/process_pools.py [hunk 3]
```diff
@@ -110,24 +111,29 @@ async def check_shim(instance_id: UUID) -> None:
...
+                starting_time_threshold = instance.started_at + timedelta(seconds=starting_timeout)
                 expire_starting = starting_time_threshold < get_current_datetime()
                 if expire_starting:
                     instance.status = InstanceStatus.TERMINATING
+                    logger.warning(
+                        "The Instance %s canot start in %s seconds. Marked as TERMINATED",
```
Comment (Egor-S) [inline src/dstack/_internal/server/background/tasks/process_pools.py]: Typo in `cannot`

File: src/dstack/_internal/server/background/tasks/process_pools.py [hunk 4]
```diff
@@ -148,8 +154,13 @@ def instance_healthcheck(*, ports: Dict[int, int]) -> HealthStatus:
                 healthy=False,
                 reason=f"Service name is {resp.service}, service version: {resp.version}",
             )
+    except requests.RequestException as e:
+        return HealthStatus(healthy=False, reason=f"Can't request shim: {e}")
     except Exception as e:
-        return HealthStatus(healthy=False, reason=f"Exception ({e.__class__.__name__}): {e}")
+        logger.exception("Unkonw exception from shim.healthcheck: %s", e)
```
Comment (Egor-S) [inline src/dstack/_internal/server/background/tasks/process_pools.py]: Typo in `Unknown`

File: src/dstack/_internal/server/background/tasks/process_pools.py [hunk 5]
```diff
@@ -148,8 +154,13 @@ def instance_healthcheck(*, ports: Dict[int, int]) -> HealthStatus:
...
+        return HealthStatus(
+            healthy=False, reason=f"Unkonw exception ({e.__class__.__name__}): {e}"
```
Comment (Egor-S) [inline src/dstack/_internal/server/background/tasks/process_pools.py]: Typo

File: src/dstack/_internal/server/routers/pools.py [hunk 6]
```diff
@@ -110,7 +110,7 @@ async def show_pool(
 async def add_instance(
     body: AddRemoteInstanceRequest,
     session: AsyncSession = Depends(get_session),
-    user_project: Tuple[UserModel, ProjectModel] = Depends(ProjectMember()),
+    user_project: Tuple[UserModel, ProjectModel] = Depends(ProjectAdmin()),
```
Comment (Egor-S) [inline src/dstack/_internal/server/routers/pools.py]: Is it an admin-level operation? Users can submit runs, but cannot create an instance beforehand. Isn't it strange?

Comment (peterschmidt85) [reply src/dstack/_internal/server/routers/pools.py]: Users should be able to manage instances

File: src/dstack/_internal/server/background/tasks/process_pools.py (hunk 1 above)
Comment (TheBits) [reply src/dstack/_internal/server/background/tasks/process_pools.py]: Good idea! Added.

File: src/dstack/_internal/server/background/tasks/process_pools.py (hunk 2 above)
Comment (TheBits) [reply src/dstack/_internal/server/background/tasks/process_pools.py]: Thanks! Added as constant.

File: src/dstack/_internal/server/background/tasks/process_pools.py (hunk 3 above)
Comment (TheBits) [reply src/dstack/_internal/server/background/tasks/process_pools.py]: Fixed

File: src/dstack/_internal/server/background/tasks/process_pools.py (hunk 5 above)
Comment (TheBits) [reply src/dstack/_internal/server/background/tasks/process_pools.py]: Thx! Fixed

File: src/dstack/_internal/server/routers/pools.py (hunk 6 above)
Comment (TheBits) [reply src/dstack/_internal/server/routers/pools.py]: Thanks. Fixed for all methods.

**Pull request merged**